python/cudf/cudf/core/column/string.py

```diff
@@ -5434,50 +5459,46 @@ def normalize_binop_value(
             and other.dtype == "object"
         ):
             return other
-        if isinstance(other, str) or other is None:
-            return utils.scalar_broadcast_to(
-                other, size=len(self), dtype="object"
-            )
-        if isinstance(other, np.ndarray) and other.ndim == 0:
-            return utils.scalar_broadcast_to(
-                other.item(), size=len(self), dtype="object"
-            )
-        raise TypeError(f"cannot broadcast {type(other)}")
+        if isinstance(other, str):
+            return cudf.Scalar(other)
+        return NotImplemented
 
-    def binary_operator(
-        self, op: str, rhs, reflect: bool = False
+    def _binaryop(
+        self, other: ColumnBinaryOperand, op: str
     ) -> "column.ColumnBase":
+        reflect, op = self._check_reflected_op(op)
+        # TODO: Try to find a way to disable these ops while still exposing
+        # this behavior. It may not be possible for now though.
```

Comment:
[As my comment here indicates](https://github.com/rapidsai/cudf/pull/10509/files/74ae0af2998f30d708d26611ed0d5832ab2f6e31#diff-81a50fe0877c20003a77288cbadef4aa11bb7cad4575063e8b3babeb0981e204R5043), you cannot actually subtract, mod, etc with string columns. However, because of https://github.com/pandas-dev/pandas/issues/46332 we need to define those operators just so that if someone does try to divide with a column of strings and the [column happens to be empty or all null we can produce the same output as pandas](https://github.com/rapidsai/cudf/pull/10509/files/74ae0af2998f30d708d26611ed0d5832ab2f6e31#diff-81a50fe0877c20003a77288cbadef4aa11bb7cad4575063e8b3babeb0981e204R5475).

How would you recommend rephrasing this comment with that context?

Reply:
Updated the comment.